### DISCUSSION TITLE
Named anchors in tree draw-node

### DISCUSSION THREAD

**Discussion by noahjutz** [cetz-package/cetz#698] [Q&A] [answered]
Is it possible to refer to a tree's node's position from outside of a tree?

When I try

```
cetz.tree.tree(
  // ...
  draw-node: (node, p) => {
    if node.content.text == "1" {circle((), radius: 0, name: "anchor_1")}
    else if node.content.text == "2" {circle((), radius: 0, name: "anchor_2")}
  }
)

line("anchor_1", "anchor_2")
```

I get `Assertion failed: Unknown element 'anchor_1' in elements ()`.

Motivation: I want to draw arrows between sibling nodes in a tree.

**Comment by johannes-wolf** [comment, marked as answer]
Currently, you cannot use custom names, but reference tree nodes via their "index" anchors: "0-0" would be the first child of the root node, "0-1" the second etc. This is not ideal, I've created a ticket to allow outputting anchors.
